src/c++/num_util/num_util.cpp

```diff
@@ -359,18 +355,24 @@ void copy_data(boost::python::numeric::array arr, char* new_data){
 } 
 
 //Return a clone of this array
-numeric::array clone(numeric::array arr){
+pyndarray clone(pyndarray arr){
   object obj(handle<>(PyArray_NewCopy((PyArrayObject*)arr.ptr(),PyArray_CORDER)));
   return makeNum(obj);
 }
 
-  
+
 //Return a clone of this array with a new type
-numeric::array astype(boost::python::numeric::array arr, PyArray_TYPES t){
-  return (numeric::array) arr.astype(type2char(t));
+#if BOOST_VERSION < 106500
```

Comment:
It's probably cleaner to move the `ifdef` into the function definition (which is the same for both cases).